server/webhook.go

```diff
@@ -230,8 +230,8 @@ func (p *Plugin) permissionToProject(ctx context.Context, userID, namespace, pro
 		return false
 	}
 
-	// Check for guest level permissions
-	if result.Permissions.ProjectAccess == nil || result.Permissions.ProjectAccess.AccessLevel == gitlabLib.GuestPermissions {
+	// Check for guest level permissions for project and group
+	if (result.Permissions.ProjectAccess != nil && result.Permissions.ProjectAccess.AccessLevel == gitlabLib.GuestPermissions) || (result.Permissions.GroupAccess != nil && result.Permissions.GroupAccess.AccessLevel == gitlabLib.GuestPermissions) {
```

Comment:
Are we sure that the issue others are running into is related to this particular guest permission check?

Also, the meaning of this condition changed a bit. The first half of this conditional is similar to before but not equivalent:

```diff
- result.Permissions.ProjectAccess == nil || result.Permissions.ProjectAccess.AccessLevel == gitlabLib.GuestPermissions
+ result.Permissions.ProjectAccess != nil && result.Permissions.ProjectAccess.AccessLevel == gitlabLib.GuestPermissions
```

What's the reasoning for this logic change? More specifically
- what does it mean for `result.Permissions.ProjectAccess` to be `nil` or not
- what does it mean for  `esult.Permissions.ProjectAccess.AccessLevel` to be  `gitlabLib.GuestPermissions` or not

Reply:
@mickmister If the repo is a part of a group then we only get the GroupAccess object and the ProjectAccess is nil in that case and vice versa. I have tried with a bunch of different cases. Both cannot be nil at the same time. 

So I have put the condition that we will check the access only when it is not nil.

Reply:
So now the condition is 
If the repo is part of the group in that case 
1) ProjectAccess will be nil so we don't check the permission in that case 
2) GroupAccess will be not nil and we check whether the user is a guest in that group or not , if it is we will return false from the function.

Reply:
I think we should go with `<=` instead of `==` here. The system of numbers GitLab uses for this is a hierarchy, where the level of access increases as the number. Using `<=` will allow us to say "If they have guest privileges or less permissions than that". We can put a comment that says this too